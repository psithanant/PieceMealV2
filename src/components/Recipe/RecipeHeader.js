import React from 'react';
import { Header, Button, Icon, Title, Body, Right, Left } from 'native-base';

function RecipeHeader(props) {
  return (
    <Header color='#F8EEE7' >
      <Left />
        <Body>
            <Title>Recipes</Title>
        </Body>
        <Right>
          <Button
            onPress={() => props.navigationHero.navigation.navigate('Home')}
            transparent
          >
            <Icon name="add" size={35} />
          </Button>
        </Right>
    </Header>
  );
}

export default RecipeHeader;
