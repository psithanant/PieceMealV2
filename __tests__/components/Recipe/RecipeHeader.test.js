import 'react-native';
import React from 'react';

import RecipeHeader from '../../../src/components/Recipe/RecipeHeader';
import renderer from 'react-test-renderer';

describe('tests the RecipeHeader component', () => {
  it('renders recipeHeader properly', () => {
    expect(renderer.create(
      <RecipeHeader
        navigate={jest.fn}
      />
    )).toMatchSnapshot();
  });
});
