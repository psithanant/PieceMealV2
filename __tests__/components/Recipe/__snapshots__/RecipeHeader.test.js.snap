// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the RecipeHeader component renders recipeHeader properly 1`] = `
<View>
  <View
    color="#F8EEE7"
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderBottomColor": "#a7a6ab",
        "borderBottomWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "center",
        "left": 0,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 15,
        "right": 0,
        "top": 0,
      }
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "flex": 1,
        }
      }
      virtual={undefined}
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
        virtual={undefined}
      >
        Recipes
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
      virtual={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": null,
            "backgroundColor": "transparent",
            "borderColor": null,
            "borderRadius": 50,
            "borderWidth": null,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginRight": -8,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 35,
              },
              Object {
                "color": "#007aff",
                "fontSize": 27,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 2,
                "paddingTop": 2,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          virtual={undefined}
        >
          ï„‚
        </Text>
      </View>
    </View>
  </View>
</View>
`;
